include/guestbook_db.inc

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+* @file
+* Implementation of database access functions.
+*/
+
+/**
+ * Reading data from the database.
+ *
+ * @return
+ *   A data array containing complete information about each database entry.
+ */
+function guestbook_read_db($table = 'guestbook_drupal', $order = 'id', $direction = 'DESC', $limit = '10') {
+  $rows = array();
+  $query = db_select($table, 't')
+    ->extend('PagerDefault')
+    ->fields('t', array('id', 'uid', 'sid', 'date', 'message'))
+    ->orderBy($order, $direction)
+    ->limit($limit)
+    ->execute();
+  while ($value = $query->fetchAssoc()) {
+    $rows[] = [
+      $value['id'],      // Message id.
+      $value['uid'],     // User id.
+      $value['sid'],     // Session id.
+      $value['date'],    // Creation date of message.
+      $value['message'], // Message text.
+    ];
+  }
+  return $rows;
+}
+
+/**
+ * Save or edit message.
+ *
+ * @param $form_state
+ *   Data from hook form. Contains $form['infoById'] = array('id', 'uid', 'sid', 'date', 'message').
+ *   id - message id; uid - user id; sid - session id; date - message creation date; message - message text.
+ */
+function guestbook_save_edit_db($form_state) {
+  global $user;
+
+  // If there is an array of data ($form['infoById']['id']), then edit the message
+  // otherwise - save as a new message.
+  if (isset ($form_state['values']['infoById']['id'])) {
+    db_update('guestbook_drupal')
+      ->fields(array('message' => $form_state['values']['message']))
+      ->condition('id', $form_state['values']['infoById']['id'])
+      ->execute();
+    drupal_goto('guestbook-page', array('query' => drupal_get_query_parameters()));
```

Comment:
this is not good practice using drupal_goto. It will avoid to alter some function. If you use this function in the submit callback use `#redirect` property for `$form_state` array instead